Add parcel flag

emmron opened the pull request:
<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)

  For any major changes, please first open a bug report (if it's a bug) or a feature request.
-->

## Description

<!-- Write a brief description of the changes introduced by this PR -->

### Documentation

<!--
  Where is this feature or API documented?

  - If docs exist:
    - Update any references, if relevant. This includes Guides and Gatsby Internals docs.
  - If no docs exist:
    - Create a stub for documentation including bullet points for how to use the feature, code snippets (including from happy path tests), etc.
-->

### Tests

<!-- Did you add tests (unit tests, E2E tests, etc.)? How did you test this change? -->

## Related Issues

<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234

  Link to an issue that is partially addressed by this PR (if there are any)
  e.g. Addresses #1234

  Link to related issues (if there are any)
  e.g. Related to #1234
-->

serhalp commented:
@emmron This PR is very large and unwieldy, updates unrelated code, and it isn't clear to me what it's solving. To be honest it looks like it may even have been AI generated, and it looks very similar to https://github.com/gatsbyjs/gatsby/pull/39042. I'm going to close it, but please feel free to open smaller, more easily reviewable PRs.

emmron commented:
@serhalp 
This pull request includes several updates across multiple files to clean up code, add new features, and improve overall functionality. The most important changes include removing commented-out code in the ESLint configuration, adding a new development container configuration, and refactoring the `gatsby-node.js` file for better error handling and code readability.

### Code Cleanup and Configuration:

* [`.eslintrc.js`](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L123-L132): Removed multiple commented-out rules and explanations to clean up the file and improve readability. [[1]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L123-L132) [[2]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L152) [[3]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L210-L223) [[4]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L233-L239) [[5]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L248-L250) [[6]](diffhunk://#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5L259-L260)
* [`.circleci/config.yml`](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L142-L150): Removed redundant cache steps in the CircleCI configuration.

### New Features:

* [`.devcontainer/devcontainer.json`](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R1-R5): Added a new development container configuration using Microsoft's universal image.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L126-R127): Added a new `develop` script to enable experimental Parcel support for Gatsby development.

### Refactoring and Improvements:

* [`packages/gatsby-plugin-manifest/src/gatsby-node.js`](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L1-R16): Refactored the file to replace `import` statements with `require`, added better error handling, and improved the manifest generation logic. [[1]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L1-R16) [[2]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L41-R33) [[3]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L56-R67) [[4]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L86-L146) [[5]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L156-L161) [[6]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L206-R253) [[7]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L246-L247) [[8]](diffhunk://#diff-c1300a9384a9c6ec6ab74c0c727cb4a0b5be188fc0ddc6f95a3b412ebb78bb56L257-R373)

### Minor Changes:

* [`.cursorignore`](diffhunk://#diff-e3d5332bf30e590dd8482d2dd1b3bb8a3fa78dc7407616eddd98587606d5b713R1): Added a comment to explain the purpose of the file.
* [`examples/hn/gatsby-config.js`](diffhunk://#diff-12215fdbc8eadb7bec58ab1a906da582405c4f1153202705acced9eace733ec2R5-R14): Enabled the `DISABLE_PARCEL_FOR_CONFIG` flag in the Gatsby configuration for the Hacker News example.

emmron reviewed (commented):
@ @

emmron reviewed (commented):
@serhalp
